R/add.R

```diff
@@ -47,26 +47,7 @@ add_row <- function(.data, ..., .before = NULL, .after = NULL) {
   df <- df[names(.data)]
 
   pos <- pos_from_before_after(.before, .after, nrow(.data))
-
-  if (pos <= 0L) {
-    out <- rbind(df, .data)
-  } else if (pos >= nrow(.data)) {
-    out <- rbind(.data, df)
-  } else {
-    indexes <- seq_len(pos)
-    out <- rbind(.data[indexes, ], df, .data[-indexes, ])
-  }
-
-  # If .data is empty, coerce the out column class to .data class
-  if (nrow(.data) == 0) {
-    for (i in 1:ncol(out)) {
-      if (class(.data[[i]])[1] == "factor") {
-        out[[i]] <- as.factor(NA)
-      } else {
-        class(out[[i]]) <- class(.data[[i]])
-      }
-    }
-  }
+  out <- rbind_safe(.data, df, pos)
```

Comment:
Thanks. Looks like you could inline rbind_position() so that all cases are handled in one function, perhaps `rbind_at()` instead of `rbind_safe()`?